include/deal.II/distributed/tria.h

```diff
@@ -943,6 +940,172 @@ namespace parallel
+         * The parameter @p handle corresponds to the position where the
+         * @p unpack_callback function is allowed to read from the memory. Its
+         * value needs to be in accordance with the corresponding pack_callback
+         * function that has been registered previously.
+         *
+         * Data has to be previously transferred with execute_transfer()
+         * or read from the file system via load().
+         */
+        void
+        unpack_data(
+          const std::vector<quadrant_cell_relation_t> &quad_cell_relations,
+          const unsigned int                           handle,
+          const std::function<void(
+            const typename dealii::Triangulation<dim, spacedim>::cell_iterator
+              &,
+            const typename dealii::Triangulation<dim, spacedim>::CellStatus &,
+            const boost::iterator_range<std::vector<char>::const_iterator>)>
+            &unpack_callback) const;
+
+        /**
+         * Transfer data to file system.
+         *
+         * The data will be written in a separate file, whose name
+         * consists of the stem @p filename and an attached identifier
+         * <tt>-fixed.data</tt>.
+         *
+         * All processors write into that file simultaneously via MPIIO.
+         * Each processor's position to write to will be determined
+         * from the provided @p parallel_forest.
+         *
+         * Data has to be previously packed with pack_data().
+         */
+        void
+        save(const typename dealii::internal::p4est::types<dim>::forest
+               *         parallel_forest,
+             const char *filename) const;
+
+        /**
+         * Transfer data from file system.
+         *
+         * The data will be read from separate file, whose name
+         * consists of the stem @p filename and an attached identifier
+         * <tt>-fixed.data</tt>. The @p n_attached_deserialize parameter
+         * is required to gather the memory offsets for each callback.
+         *
+         * All processors read from that file simultaneously via MPIIO.
+         * Each processor's position to read from will be determined
+         * from the provided @p parallel_forest.
+         *
+         * After loading, unpack_data() needs to be called to finally
+         * distribute data across the associated triangulation.
+         */
+        void
+        load(const typename dealii::internal::p4est::types<dim>::forest
+               *                parallel_forest,
+             const char *       filename,
+             const unsigned int n_attached_deserialize);
+
+        /**
+         * Maximum number of callback functions that have been
+         * called for packing data over all cells.
+         *
+         * Returns the size of the class member cumulative_sizes_fix.
+         * Data has to be previously packed with pack_data() to get
+         * a non-zero result.
+         */
+        unsigned int
+        num_packed_datas();
```

Comment:
use `n_packaged_data`

Reply:
I used this function solely for an assertion somewhere, which I cannot find anymore. I guess we do not need it anymore, so I'll remove it.

Reply:
That's one solution :-)